readium/shared/src/main/java/org/readium/r2/shared/resource/Container.kt

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright 2023 Readium Foundation. All rights reserved.
+ * Use of this source code is governed by the BSD-style license
+ * available in the top-level LICENSE file of the project.
+ */
+
+package org.readium.r2.shared.resource
+
+import java.io.File
+import org.readium.r2.shared.util.SuspendingCloseable
+
+/**
+ * A resource container as an archive or a directory.
+ */
+interface Container : SuspendingCloseable {
+
+    /**
+     * Holds a container entry's.
+     */
+    interface Entry : Resource {
+
+        /**
+         * Absolute path to the entry in the archive.
+         * It MUST start with /.
+         */
+        val path: String
+
+        /**
+         * Gets the entry name if any.
+         */
+        override suspend fun name(): ResourceTry<String?>
```

Comment:
What is being overridden here? It looks like the same signature as in `Resource`.

And what is the difference with `path` in the context of a `Container.Entry`?

Reply:
I'm pretty sure I overrode `name` to get rid of complaints from the compiler about some ambiguity. I can't reproduce that though, it might be outdated.
`path` is a complete file path which can reflect some hierarchical organisation. `name` is just a name.